servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/TaskBasedAsyncCompletableOperator.java

```diff
@@ -292,27 +314,42 @@ public void onError(final Throwable t) {
      */
     static final class OffloadedCancellable implements Cancellable {
         private final Cancellable cancellable;
+        private final BooleanSupplier shouldOffload;
         private final Executor executor;
 
-        OffloadedCancellable(final Cancellable cancellable, final Executor executor) {
+        OffloadedCancellable(final Cancellable cancellable,
+                             final BooleanSupplier shouldOffload, final Executor executor) {
             this.cancellable = requireNonNull(cancellable);
+            this.shouldOffload = shouldOffload;
             this.executor = executor;
         }
 
         @Override
         public void cancel() {
+            boolean offloadCancel;
             try {
-                LOGGER.debug("executing Completable cancel on {}", executor);
-                executor.execute(() -> safeCancel(cancellable));
+                offloadCancel = shouldOffload.getAsBoolean();
             } catch (Throwable t) {
-                LOGGER.warn("Failed to execute task on the executor {}. " +
```

Comment:
we should still log a warning as throwing isn't supported and we make an assumption about offloading that may have impacts on the application.

Reply:
Added